x/feedist/model_test.go

```diff
@@ -0,0 +1,71 @@
+package feedist
+
+import (
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+func TestRevenueValidate(t *testing.T) {
+	addr := weave.Address("f427d624ed29c1fae0e2")
+
+	cases := map[string]struct {
+		model   Revenue
+		wantErr error
+	}{
+		"valid model": {
+			model: Revenue{
+				Admin: addr,
+				Recipients: []*Recipient{
+					{Weight: 1, Address: addr},
+				},
+			},
+			wantErr: nil,
+		},
+		"admin address must be present": {
+			model: Revenue{
+				Admin: nil,
+				Recipients: []*Recipient{
+					{Weight: 1, Address: addr},
+				},
+			},
+			wantErr: errors.InvalidModelErr,
+		},
+		"at least one recipient must be given": {
+			model: Revenue{
+				Admin:      addr,
+				Recipients: []*Recipient{},
+			},
+			wantErr: errors.InvalidModelErr,
+		},
+		"recipient weight must be greater than zero": {
+			model: Revenue{
+				Admin: addr,
+				Recipients: []*Recipient{
+					{Weight: 0, Address: addr},
+				},
+			},
+			wantErr: errors.InvalidModelErr,
+		},
+		"recipient must have a valid address": {
```

Comment:
I like these labels for each test case...
I always used slices with indexes, but this map with meaningful strings is very nice to read